clang/include/clang/Driver/Options.td

```diff
@@ -6167,56 +6174,43 @@ def dump_depscan_tree_EQ : Option<["-"], "dump-depscan-tree=", KIND_JOINED>,
-    HelpText<"Path to a persistent on-disk backing store for the builtin CAS.">,
-    ShouldParseIf<!strconcat(fcas.KeyPath," == CASOptions::BuiltinCAS")>,
+    HelpText<"Path to a persistent on-disk backing store for the builtin CAS."
+             " Use '-' to have the compiler use a default location.">,
     MarshallingInfoString<CASOpts<"BuiltinPath">>;
 
-def fcas_plugin_path : Separate<["-"], "fcas-plugin-path">,
-    Group<f_Group>, MetaVarName<"<dsopath>">,
-    HelpText<"Path to the CAS plugin to load.">,
-    ShouldParseIf<!strconcat(fcas.KeyPath," == CASOptions::PluginCAS")>,
-    MarshallingInfoString<CASOpts<"PluginPath">>;
-
-def fcas_plugin_arg : Joined<["-"], "fcas-plugin-arg">,
-    Group<f_Group>, MetaVarName<"<option>">,
-    HelpText<"Arguments for initializing the CAS plugin.">,
-    ShouldParseIf<!strconcat(fcas.KeyPath," == CASOptions::PluginCAS")>,
-    MarshallingInfoStringVector<CASOpts<"PluginArgs">>;
-
 def fcas_fs : Separate<["-"], "fcas-fs">,
     Group<f_Group>, MetaVarName<"<tree>">,
-    HelpText<"Configure the filesystem to read from the provided CAS tree.">,
-    ShouldParseIf<!strconcat(fcas.KeyPath, " != CASOptions::NoCAS")>,
-    MarshallingInfoString<CASOpts<"CASFileSystemRootID">>;
+    HelpText<"Configure the filesystem to read from the provided CAS tree."
+             " See also -fcas-builtin-path for loading a tree.">,
+    MarshallingInfoString<FileSystemOpts<"CASFileSystemRootID">>;
 
 def fcas_fs_working_directory : Separate<["-"], "fcas-fs-working-directory">,
     Group<f_Group>, MetaVarName<"<dir>">,
     HelpText<"Working directory for -fcas-fs (if not the root).">,
     ShouldParseIf<!strconcat("!", fcas_fs.KeyPath, ".empty()")>,
-    MarshallingInfoString<CASOpts<"CASFileSystemWorkingDirectory">>;
+    MarshallingInfoString<FileSystemOpts<"CASFileSystemWorkingDirectory">>;
 
-defm fcas_fs_cache_results : BoolFOption<"cas-fs-result-cache",
-    CASOpts<"CASFileSystemResultCache">, DefaultFalse,
-    PosFlag<SetTrue, [], "When using -fcas-fs, use a result cache.">,
+defm experimental_cache_compile : BoolFOption<"experimental-cache-compile",
+    FrontendOpts<"CacheCompile">, DefaultFalse,
+    PosFlag<SetTrue, [], "Try to cache full compilation (-cc1). Only works with"
+                         " -fcas-fs.">,
     NegFlag<SetFalse>>;
 
-defm fcas_token_cache : BoolFOption<"cas-token-cache",
-    CASOpts<"CASTokenCache">, DefaultFalse,
-    PosFlag<SetTrue, [], "When reading from a CAS, cache tokens.">,
+defm experimental_cache_lex_raw : BoolFOption<"experimental-cache-lex-raw",
+    PreprocessorOpts<"CacheLexRaw">, DefaultFalse,
+    PosFlag<SetTrue, [], "When reading from a CAS, cache raw lexing.">,
     NegFlag<SetFalse>>;
 
-defm fcasid_output : BoolFOption<"casid-output",
-    CASOpts<"WriteOutputAsCASID">, DefaultFalse,
-    PosFlag<SetTrue, [], "When using -fcas-fs-result-cache, write a CASID for the output file.">,
+def fcas_token_cache : Flag<["-"], "fcas-token-cache">,
```

Comment:
Because this `-cc1` is used by the CMake options. None of the others is. Thought I'd do it in stages:
- Introduce new function leave behind alias.
- Wait for that to be around for people (us) to have time to switch to a toolchain that has the new option.
- Change CMake to use the new option and remove the alias.